simplex/block_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/StephenButtolph/canoto"
+	"github.com/ava-labs/simplex"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman/snowmantest"
+	"github.com/ava-labs/avalanchego/snow/engine/enginetest"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block/blocktest"
+)
+
+func TestBlockSerialization(t *testing.T) {
+	unexpectedBlockBytes := errors.New("unexpected block bytes")
+
+	testVM := &blocktest.VM{
+		VM: enginetest.VM{
+			T: t,
+		},
+	}
+
+	testBlock := snowmantest.Block{
+		HeightV: 1,
+		BytesV:  []byte("test block"),
+	}
+
+	b := &Block{
+		vmBlock: &testBlock,
+		metadata: simplex.ProtocolMetadata{
+			Version: 1,
+			Epoch:   1,
+			Round:   1,
+			Seq:     1,
+			Prev:    [32]byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07},
+		},
+	}
+
+	// Serialize the block
+	blockBytes, err := b.Bytes()
+	require.NoError(t, err)
+
+	tests := []struct {
+		name          string
+		expected      []byte
+		parseFunc     func(context.Context, []byte) (snowman.Block, error)
+		expectedError error
+		blockBytes    []byte
+	}{
+		{
+			name:       "block serialization",
+			blockBytes: blockBytes,
+			expected:   blockBytes,
+			parseFunc: func(_ context.Context, b []byte) (snowman.Block, error) {
+				require.Equal(t, b, testBlock.BytesV, "block bytes should match")
```

Comment:
`Equal` takes in `(expected, actual)`.
```suggestion
				require.Equal(t, testBlock.BytesV, b, "block bytes should match")
```